Library/Homebrew/utils/shell.rb

```diff
@@ -90,6 +85,28 @@ def prepend_path_in_shell_profile(path)
         "echo 'set -g fish_user_paths \"#{sh_quote(path)}\" $fish_user_paths' >> #{shell_profile}"
       end
     end
-    module_function :prepend_path_in_shell_profile
+
+    # cut the actual value of ARG_MAX in half for safety since
+    # environment variables count toward the total
+    def max_command_length
+      Utils.popen_read("/usr/bin/getconf", "ARG_MAX").to_i / 2
```

Comment:
I think halving the `ARG_MAX` is kinda radical.

```ruby
ENV.inject(0) { |c, (k, v)| c + 2 + k.length + v.length }
```

get's the exact length of the environment variables. Doubling that should be enough safety margin.